### PR TITLE
Allow services to be merged when disallowing all duplicates

### PR DESCRIPTION
**Pull request by kageiit**
Services are treated specially just like manifest during merging and should be excluded from duplicate disallow logic

### CODE REVIEW

File: test/com/facebook/buck/jvm/java/JarBuilderTest.java [hunk 1]
```diff
@@ -85,53 +86,57 @@ public void testSortsEntriesFromAllContainers() throws IOException {
...
-                  .addEntry("META-INF/services/com.example.Foo2", "com.example.Bar4"))
-          .addEntryContainer(
-              container3
-                  .addEntry("META-INF/services/com.example.Foo2", "com.example.Bar3")
-                  .addEntry("META-INF/services/foo/bar", "bar"))
-          .createJarFile(tempFile.toPath());
-    }
 
-    try (JarFile jarFile = new JarFile(tempFile)) {
+    for (boolean shouldDisallowAllDuplicates : Arrays.asList(true, false)) {
```
Comment (ttsugriy) [inline test/com/facebook/buck/jvm/java/JarBuilderTest.java]: nit: how about parameterized tests? https://github.com/junit-team/junit4/wiki/parameterized-tests :)

Comment (kageiit) [reply test/com/facebook/buck/jvm/java/JarBuilderTest.java]: Sure. I can make the change

**Comment by ttsugriy** [comment]
sorry about the delay. Looks like ant pmd is failing with
```
test/com/facebook/buck/jvm/java/JarBuilderTest.java:51:	This class name ends with Test but contains no test cases
```

You should be able to reproduce it with `ant pmd`. The problem is that this seems to be a bug in PMD, so I was trying to update PMD and check if it helps, but unfortunately even the latest version of PMD fails with the same error :(

**Comment by ttsugriy** [comment]
ok, I gave up and reverted to using plain for loop instead of parameterized tests :( With JUnit5 it would be so much nicer... Sigh...

**Comment by kageiit** [comment]
> ok, I gave up and reverted to using plain for loop instead of parameterized tests :( With JUnit5 it would be so much nicer... Sigh...

ouch. sorry to hear. Thanks for working around it though

**Comment by ttsugriy** [comment]
no worries, @kageiit. I just felt bad for asking you to convert from for loop to parameterized test just to discover that it does not work :)

**Pull request closed without merging**